fxrunner/src/lib/splash.rs

```diff
@@ -0,0 +1,252 @@
+        })
+        .map(drop)?;
+
+        self.ui_thread_join_handle
+            .take()
+            .expect("Splash::destroy called without UI thread")
+            .join()
+            .expect("UI thread panicked");
+
+        Ok(())
+    }
+}
+
+impl Drop for WindowsSplash {
+    fn drop(&mut self) {
+        assert!(
+            self.ui_thread_join_handle.is_none(),
+            "Splash dropped without calling destroy()"
+        );
+    }
+}
+
+/// Register the window class that `Splash` will use.
+///
+/// It will only attempt to register the window class if it has not yet been
+/// registered.
+fn ensure_window_class_registered(instance: &ModuleHandle) -> Result<(), io::Error> {
+    let mut cls = WNDCLASSA::default();
+
+    let exists = {
+        let rv = unsafe {
+            winuser::GetClassInfoA(
+                instance.as_ptr(),
+                WINDOW_CLASS_NAME.as_ptr(),
+                &mut cls as *mut WNDCLASSA,
+            )
+        };
+
+        if rv == 0 {
+            let err = io::Error::last_os_error();
+            if err.raw_os_error().unwrap() != winerror::ERROR_CLASS_DOES_NOT_EXIST as i32 {
+                return Err(err);
+            }
+            false
+        } else {
+            true
+        }
+    };
+
+    if exists {
+        return Ok(());
+    }
+
+    // This handle does not need to be freed.
+    let brush = check_nonnull(unsafe { wingdi::CreateSolidBrush(wingdi::RGB(255, 0, 0)) })?;
+
+    // This handle does not need to be freed.
+    let cursor = check_nonnull(unsafe { winuser::LoadCursorW(null_mut(), winuser::IDC_ARROW) })?;
+
+    cls.style = 0;
+    cls.lpfnWndProc = Some(window_proc);
+    cls.cbClsExtra = 0;
+    cls.cbWndExtra = 0;
+    cls.hInstance = instance.as_ptr();
+    cls.hIcon = null_mut();
+    cls.hCursor = cursor;
+    cls.hbrBackground = brush;
+    cls.lpszMenuName = null_mut();
+    cls.lpszClassName = WINDOW_CLASS_NAME.as_ptr();
+
+    check_nonzero(unsafe { winuser::RegisterClassA(&cls as *const WNDCLASSA) }).map(drop)
+}
+
+/// Create and show a window of the given size.
+fn create_and_show_window(display_width: u32, display_height: u32) -> Result<HWND, io::Error> {
+    let instance = get_module_handle()?;
+
+    ensure_window_class_registered(&instance)?;
+
+    let window_handle = check_nonnull(unsafe {
+        winuser::CreateWindowExA(
+            winuser::WS_EX_NOACTIVATE,
+            WINDOW_CLASS_NAME.as_ptr(),
+            // We re-use the class name as the window name. There is no
+            // title bar, so it is not displayed on screen.
+            WINDOW_CLASS_NAME.as_ptr(),
+            0,
+            0,
+            0,
+            display_width as i32,
+            display_height as i32,
+            null_mut(), // No parent window.
+            null_mut(), // No menu.
+            instance.as_ptr(),
+            null_mut(),
+        )
+    })?;
+
+    // Remove the title bar.
```

Comment:
Have you tried just removing the title bar by using the `WS_POPUP` style in `dwStyle`?